File: python/docs_code_snippets/call.py
from pydantic import BaseModel
from opperai import Opper

opper = Opper()


class CapitalOutput(BaseModel):
    capital: str
    population: int | None = None


# --- docs ---
result = opper.call(
    "get_capital",
    instructions="Return the capital of the given country.",
    input={"country": "Sweden"},
    output_schema=CapitalOutput,
)

print(result.data.capital)
# --- /docs ---

File: typescript/docs_code_snippets/call.ts
import { z } from "zod";
import { Opper } from "opperai";

const opper = new Opper();

const result = await opper.call("get_capital", {
  instructions: "Return the capital of the given country.",
  input: { country: "Sweden" },
  output_schema: z.object({
    capital: z.string(),
    population: z.number().optional(),
  }),
});

console.log(result.data.capital);
